pubsub-protocol-binding.md

```diff
@@ -0,0 +1,276 @@
+
+### 1.2. Relation to Pub/Sub
+
+Events can be produced using a Topic or consumed using Push or Pull based Subscriptions. 
+
+### 1.3. Content Modes
+
+The specification defines two content modes for transferring events:
+_structured_ and _binary_.
+
+In the _structured_ content mode, event metadata attributes and event data are
+placed into the Pub/Sub message body section using an
+[event format](#14-event-formats).
+
+In the _binary_ content mode, the value of the event `data` attribute is placed
+into the Pub/Sub payload data as-is, with the `datacontenttype`
+attribute value declaring its media type; all other event attributes are mapped
+to Pub/Sub attributes.
+
+### 1.4. Event Formats
+
+Event formats, used with the _structured_ content mode, define how an event is
+expressed in a particular data format. All implementations of this
+specification MUST support the [JSON event format][json-format], but MAY support
+any additional, including proprietary, formats.
+
+### 1.5. Security
+
+This specification does not introduce any new security features for Pub/Sub, or
+mandate specific existing features to be used.
+
+## 2. Use of CloudEvents Attributes
+
+This specification does not further define any of the [CloudEvents][ce] event
+attributes.
+
+This mapping is intentionally robust against changes, including the addition and
+removal of event attributes, and also accommodates vendor extensions to the
+event metadata.
+
+### 2.1. datacontenttype Attribute
+
+The `datacontenttype` attribute is assumed to contain a [RFC2046][rfc2046]
+compliant media-type expression.
+
+### 2.2. data Attribute
+
+The `data` attribute is assumed to contain opaque application data that is
+encoded as declared by the `datacontenttype` attribute.
+
+An application is free to hold the information in any in-memory representation
+of its choosing, but as the value is transposed into a Pub/Sub message as defined 
+in this specification, the assumption is that the `data` attribute value is made
+available as a sequence of bytes.
+
+For instance, if the declared `datacontenttype` is
+`application/json;charset=utf-8`, the expectation is that the `data` attribute
+value is made available as [UTF-8][rfc3629] encoded JSON text Pub/Sub message data.
+
+## 3. Pub/Sub Message Mapping
+
+With Pub/Sub, the content mode is chosen by the sender of the
+event. Protocol usage patterns that might allow solicitation of events using a
+particular content mode might be defined by an application, but are not defined
+here.
+
+The receiver of the event can distinguish between the two content modes by
+inspecting the [Pub/Sub Attribute `Content-Type`][content-type] of the
+[Pub/Sub message][pubsub-message]. If the attribute is present and its value is prefixed 
+with the CloudEvents media type `application/cloudevents`, indicating the use of a known
+[event format](#14-event-formats), the receiver uses _structured_ mode,
+otherwise it defaults to _binary_ mode.
```

Comment:
"the receiver uses": Is this guidance to the implementor or a statement of fact? This should use RFC2119 words.

Reply:
rephrased to: `the receiver is expected to use _structured_ mode,
otherwise it SHOULD default to _binary_ mode.` WDYT?